subsys/usb/host/class/usbh_hub_mgr.c

```diff
@@ -0,0 +1,1166 @@
+					k_mutex_unlock(&hub_mgr_data->lock);
+					
+					k_work_submit(&hub_mgr_data->port_work.work);
+					return;
+				}
+			}
+		}
+	}
+
+	k_mutex_unlock(&hub_mgr_data->lock);
+
+	if (!hub_mgr_data->int_active && !hub_mgr_data->being_removed) {
+		usbh_hub_mgr_start_interrupt(hub_mgr_data);
+	}
+}
+
+/**
+ * @brief Hub interrupt IN callback
+ */
+static int usbh_hub_mgr_interrupt_in_cb(struct usb_device *const dev,
+					 struct uhc_transfer *const xfer)
+{
+	struct usbh_hub_mgr_data *hub_mgr_data = (struct usbh_hub_mgr_data *)xfer->priv;
+	struct net_buf *buf = xfer->buf;
+
+	if (!hub_mgr_data || hub_mgr_data->being_removed) {
+		goto cleanup_and_exit;
+	}
+
+	hub_mgr_data->int_active = false;
+
+	hub_mgr_data->prime_status = HUB_PRIME_NONE;
+
+	if (!buf || buf->len == 0) {
+		LOG_ERR("Hub level %d interrupt transfer failed or no data", 
+		hub_mgr_data->hub_level);
+		hub_mgr.processing_hub = NULL;
+		hub_mgr_data->current_port = 0;
+		goto resubmit;
+	}
+
+	memcpy(hub_mgr_data->int_buffer, buf->data, 
+	       MIN(buf->len, sizeof(hub_mgr_data->int_buffer)));
+
+	LOG_DBG("Hub level %d interrupt data received: length=%d", 
+		hub_mgr_data->hub_level, buf->len);
+
+	usbh_hub_mgr_process_data(hub_mgr_data);
+
+	net_buf_unref(buf);
+	return 0;
+
+resubmit:
+	/* Resubmit transfer */
+	if (hub_mgr_data->connected && hub_mgr_data->state == HUB_STATE_OPERATIONAL) {
+		int ret = usbh_hub_mgr_resubmit_interrupt_in(hub_mgr_data, xfer);
+		if (ret) {
+			LOG_ERR("Failed to resubmit interrupt transfer: %d", ret);
+		}
+	} else {
+		usbh_xfer_free(hub_mgr_data->hub_udev, xfer);
+	}
+	return 0;
+
+cleanup_and_exit:
+	/* Cleanup resources */
+	if (buf) {
+		net_buf_unref(buf);
+	}
+	usbh_xfer_free(dev, xfer);
+	return 0;
+}
+
+/**
+ * @brief Hub process state machine
+ */
+static void usbh_hub_mgr_process(struct k_work *work)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct usbh_hub_mgr_data *hub_mgr_data = 
+		CONTAINER_OF(dwork, struct usbh_hub_mgr_data, hub_work);
+	struct usb_hub_descriptor *hub_desc;
+	uint8_t need_prime_interrupt = 0;
+	uint8_t process_success = 0;
+	uint16_t port_num;
+	int ret;
+
+	k_mutex_lock(&hub_mgr_data->lock, K_FOREVER);
+
+	if (hub_mgr_data->being_removed) {
+		k_mutex_unlock(&hub_mgr_data->lock);
+		return;
+	}
+
+	switch (hub_mgr_data->hub_status) {
+	case HUB_RUN_IDLE:
+		/* Hub idle, process hub status change */
+		if (hub_mgr_data->prime_status == HUB_PRIME_CONTROL) {
+			LOG_DBG("Processing Hub status change");
+			hub_mgr_data->hub_status = HUB_RUN_CLEAR_DONE;
+			process_success = 1;
+			need_prime_interrupt = 1;
+		}
+		break;
+
+	case HUB_RUN_INVALID:
+		LOG_ERR("Hub in invalid state");
+		hub_mgr_data->state = HUB_STATE_ERROR;
+		break;
+
+	case HUB_RUN_WAIT_SET_INTERFACE:
+		hub_mgr_data->hub_status = HUB_RUN_GET_DESCRIPTOR_7;
+		/* Get basic hub descriptor */
+		ret = usbh_hub_get_descriptor(&hub_mgr_data->hub_instance,
+					      hub_mgr_data->hub_instance.hub_desc_buf, 7);
```

Comment:
Instead of hardcoding 7, would it be possible to use `sizeof(struct usb_hub_descriptor)`? This implies defining the `struct usb_hub_descriptor_short` without the `DeviceRemovable` field, and maybe a `struct usb_hub_descriptor_long` with the `DeviceRemovable` flexible array member, so that `sizeof(usb_hub_descriptor)` can be used.